pywbem_mock/_dmtf_schema.py

```diff
@@ -0,0 +1,392 @@
+    that can be used by pywbem_mock to create a class and qualifier declaration
+    repository.
+
+    It also provides tools to build repositories with DMTF schema.
+    """
+    def __init__(self, schema_ver, schema_dir, use_experimental=False,
+                 verbose=False):
+        """
+        The constructor downloads the schema if the 'schema_dir' does not exist
+        and installs it into the directory defined by the 'schema_dir'
+        attribute.
+
+        If the schema is already installed it simply sets the property values so
+        that the schema can be used.
+
+        If the schema was previously downloaded but not expanded, it is
+        automatically expanded (i.e. the schema/mof subdirectory is created
+        and the mof files expanded into this directory.). This allows only
+        the single file 'schema_mof_filename' to be persisted and the expansion
+        to occur on usage which saves downloading from the DMTF repeatedly and
+        saves disk space.
+
+        Parameters:
+
+          schema_dir (:term:`string`):
+            Directory into which the DMTF schema is installed or will be
+            installed.
+
+          schema_ver  (tuple of 3 integers (x, y, z):
+            Represents the DMTF the schema version in the format x.yy.z where
+                x is the major version
+                y is the minor version
+                z is the revisions
+            This MUST represent a DMTF schema that is available from the
+            DMTF web site.
+
+          use_experimental (:class:`py:bool`):
+            If True the expermintal version of the defined schema is
+            installed.
+
+            If False (the default) the Final released version of the DMTF
+            is installed.
+
+          verbose (:class:`py:bool`):
+            If 'True' status messages are output to the console
+
+        Raises:
+            ValueError if the schema cannon be retrieved from the DMTF web site.
+            ValueError or if the 'schema_dir' differs from the version of the
+            schema currently set into the 'schema_dir'
+
+        """
+        if not isinstance(schema_ver, tuple):
+            raise ValueError('schema_ver must be tuple not %s' % schema_ver)
+        if len(schema_ver) != 3:
+            raise ValueError('DMTF Schema must have 3 integer components, '
+                             '(major, minor, rev) not %s' % schema_ver)
+        schema_type = 'Experimental' if use_experimental else 'Final'
+        self._schema_ver = schema_ver
+        self._schema_dir = schema_dir
+
+        mof_dir = 'mof%s%s' % (schema_type, self.schema_ver_str)
+        self._schema_mof_dir = os.path.join(self._schema_dir, mof_dir)
+
+        self._dmtf_schema_ver = 'cim_schema_%s' % (self.schema_ver_str)
+        self._mof_zip_bn = '%s%s-MOFs.zip' % (self._dmtf_schema_ver,
+                                              schema_type)
```

Comment:
`_dmtf_schema_ver` is the version tuple, that cannot really work.

Reply:
My understanding is also that %s on a tuple `(1, 2, 3)` results in `"(1, 2, 3)"`. I did not intend to question that. My reading of the code was that the resulting string `"(1, 2, 3)"` was used in the file name, but apparently I had an eye glitch or something. All fine.
DONE